### ISSUE TITLE
Error - Cannot read properties of undefined (reading 'prototype')

### ISSUE THREAD

**Issue by YoguiSan** [Automattic/mongoose#13015]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.9.1

### Node.js version

16.14.0

### MongoDB server version

6.0.4

### Typescript version (if applicable)

4.9.5

### Description

I'm running an Express server with Mongoose and Typescript. I followed some tutorials, but when I try connecting I get the following error:

```
D:\git\Pessoal\SCOPE\backend-node\node_modules\mongoose\lib\drivers\node-mongodb-native\collection.js:254
for (const key of Object.getOwnPropertyNames(Collection.prototype)) {
                                                        ^
TypeError: Cannot read properties of undefined (reading 'prototype')
    at Object.<anonymous> (D:\git\Pessoal\SCOPE\backend-node\node_modules\mongoose\lib\drivers\node-mongodb-native\collection.js:254:57)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Object.require.extensions.<computed> [as .js] (D:\git\Pessoal\SCOPE\backend-node\node_modules\ts-node\src\index.ts:1608:43)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (D:\git\Pessoal\SCOPE\backend-node\node_modules\mongoose\lib\drivers\node-mongodb-native\index.js:8:22)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
```
Running through Nodemon. Tried downgrading and changing Mongoose and Typescript versions, but to no avail.

### Steps to Reproduce

- Setup an Express server
- Setup Typescript and Nodemon to run .ts
- Setup MongoDB
- Install Mongoose
- Create a schema/model and try connecting with Mongoose

**Comment by hasezoey** [comment]
can you you provide a code example of what you try to call? (or better a reproduction repository?)

**Comment by YoguiSan** [comment]
Here's my repo
https://bitbucket.org/yoguisan/s.c.o.p.e.-backend-node.js
Created a branch called **bug/mongoose** containing the changes I made

**Comment by hasezoey** [comment]
i personally dont see any problem with the usage of mongoose in your project. maybe try to completely remove `node_modules` and re-install everything?

as a aside, you have `@types/mongoose` installed, which is not required anymore since mongoose 5.11 (or 5.10, cant remember)

also i could not run your project, because it tries to run a `.ts` file directly, without any compilation beforehand or a direct runner like `ts-node` (same for test / jest) and the docker version also does not seem to run properly (i have setup a `.env` file)

PS: the branch is actually called `bug/mongooser` from what i can tell

**Comment by YoguiSan** [comment]
I have tried removing node_modules and package-lock.json and reinstalling, but it didn't solve the issue.

I forgot to mention that I was using the **dev** script instead of _start_ to run the application since I'm using Nodemon, sorry about that. About the .env file, it's mostly for URLs and auth tokens, the sample values in .env.example despite not being the actual ones should be enough to prevent the application from throwing errors.

The branch name must have been a typo, though for me it appears correct.

**Comment by IslandRhythms** [comment]
Attempting to run `npm install` is getting me a bunch `code ERESOLVE`
```
 npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: express-graphql@0.12.0
npm ERR! Found: graphql@16.6.0
npm ERR! node_modules/graphql
npm ERR!   graphql@"^16.6.0" from the root project
npm ERR!   peer graphql@"14.x || 15.x || 16.x" from @apollo/cache-control-types@1.0.2
npm ERR!   node_modules/@apollo/cache-control-types
npm ERR!     @apollo/cache-control-types@"^1.0.2" from @apollo/server@4.3.3
npm ERR!     node_modules/@apollo/server
npm ERR!       @apollo/server@"^4.3.2" from the root project
npm ERR!   13 more (@apollo/server, @apollo/server-gateway-interface, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer graphql@"^14.7.0 || ^15.3.0" from express-graphql@0.12.0
npm ERR! node_modules/express-graphql
npm ERR!   express-graphql@"^0.12.0" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: graphql@15.8.0
npm ERR! node_modules/graphql
npm ERR!   peer graphql@"^14.7.0 || ^15.3.0" from express-graphql@0.12.0
npm ERR!   node_modules/express-graphql
npm ERR!     express-graphql@"^0.12.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```
See if resolving that fixes your issue

**Comment by YoguiSan** [comment]
Sorry for taking long to answer. This was caused by a dependency that was no longer in use (express-graphql), removing it didn't help.

**Comment by IslandRhythms** [comment]
Do you experience this issue on your machine when running `npm run dev`?
```
C:\Users\Drumm\Desktop\Work\debugging\repos\yoguisan-s.c.o.p.e.-backend-node.js-6372e49530f7\utils\mysql.ts:43
const isLocal = ENV.toLowerCase() === 'local';
                    ^
TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at Object.<anonymous> (C:\Users\Drumm\Desktop\Work\debugging\repos\yoguisan-s.c.o.p.e.-backend-node.js-6372e49530f7\utils\mysql.ts:43:21)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Module.m._compile (C:\Users\Drumm\Desktop\Work\debugging\repos\yoguisan-s.c.o.p.e.-backend-node.js-6372e49530f7\node_modules\ts-node\src\index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Object.require.extensions.<computed> [as .ts] (C:\Users\Drumm\Desktop\Work\debugging\repos\yoguisan-s.c.o.p.e.-backend-node.js-6372e49530f7\node_modules\ts-node\src\index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\Drumm\Desktop\Work\debugging\repos\yoguisan-s.c.o.p.e.-backend-node.js-6372e49530f7\services\operador.ts:1:1)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by barclayiversen** [comment]
Sorry to re-open a closed issue. But I'm running into this problem and have code to reproduce the issue. 

https://github.com/barclayiversen/mongoose-issue-repro

To reproduce:
make a POST request to http://localhost:3000/api/products



stack trace

`
error - node_modules\mongoose\lib\types\objectid.js (21:31) @ prototype
error - TypeError: Cannot read properties of undefined (reading 'prototype')
    at eval (webpack-internal:///(sc_server)/./node_modules/mongoose/lib/types/objectid.js:15:36)
    at (sc_server)/./node_modules/mongoose/lib/types/objectid.js (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:2761:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js:7:18)
    at (sc_server)/./node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:484:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./node_modules/mongoose/lib/drivers/node-mongodb-native/index.js:4:22)
    at (sc_server)/./node_modules/mongoose/lib/drivers/node-mongodb-native/index.js (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:506:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./node_modules/mongoose/lib/index.js:4:100)
    at (sc_server)/./node_modules/mongoose/lib/index.js (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:2090:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./node_modules/mongoose/index.js:5:18)
    at (sc_server)/./node_modules/mongoose/index.js (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:275:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./backend/config/dbConnect.js:5:66)
    at (sc_server)/./backend/config/dbConnect.js (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:176:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./app/api/products/route.js:7:83)
    at (sc_server)/./app/api/products/route.js (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:165:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./node_modules/next/dist/build/webpack/loaders/next-app-loader.js?name=app%2Fapi%2Fproducts%2Froute&appPaths=&pagePath=private-next-app-dir%2Fapi%2Fproducts%2Froute.js&appDir=C%3A%5CUsers%5Cspracto%5Csrc%5CnestPracticeProject%5Cnext-practice-project-js%5Capp&pageExtensions=tsx&pageExtensions=ts&pageExtensions=jsx&pageExtensions=js&rootDir=C%3A%5CUsers%5Cspracto%5Csrc%5CnestPracticeProject%5Cnext-practice-project-js&isDev=true&tsconfigPath=tsconfig.json&assetPrefix=!:13:148)
    at (sc_server)/./node_modules/next/dist/build/webpack/loaders/next-app-loader.js?name=app%2Fapi%2Fproducts%2Froute&appPaths=&pagePath=private-next-app-dir%2Fapi%2Fproducts%2Froute.js&appDir=C%3A%5CUsers%5Cspracto%5Csrc%5CnestPracticeProject%5Cnext-practice-project-js%5Capp&pageExtensions=tsx&pageExtensions=ts&pageExtensions=jsx&pageExtensions=js&rootDir=C%3A%5CUsers%5Cspracto%5Csrc%5CnestPracticeProject%5Cnext-practice-project-js&isDev=true&tsconfigPath=tsconfig.json&assetPrefix=! (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:154:1)
    at __webpack_require__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\webpack-runtime.js:33:43)
    at __webpack_exec__ (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:3121:39)
    at C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:3122:28
    at Object.<anonymous> (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\.next\server\app\api\products\route.js:3125:3)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.requirePage (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\node_modules\next\dist\server\require.js:88:12)
    at C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\node_modules\next\dist\server\load-components.js:49:73
    at async Object.loadComponentsImpl [as loadComponents] (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\node_modules\next\dist\server\load-components.js:49:26)
    at async DevServer.findPageComponentsImpl (C:\Users\spracto\src\nestPracticeProject\next-practice-project-js\node_modules\next\dist\server\next-server.js:600:36) {
  page: '/api/products'
}
null
`

**Comment by IslandRhythms** [comment]
Open a new issue and follow the template please.
